lib/vscode/src/vs/server/node/channel.ts

```diff
@@ -681,7 +682,7 @@ export class TerminalProviderChannel implements IServerChannel<RemoteAgentConnec
 		const resolverService = new VariableResolverService(remoteAuthority, args, process.env as platform.IProcessEnvironment);
 		const resolver = terminalEnvironment.createVariableResolver(activeWorkspace, resolverService);
 
-		const getDefaultShellAndArgs = (): { executable: string; args: string[] | string } => {
+		const getDefaultShellAndArgs = async (): Promise<{ executable: string; args: string[] | string }> => {
```

Comment:
Now that this is a promise I think we need  an `await` where it's used below.

Reply:
Ah well there is a type later down but all the arguments are optional so it won't catch this.